src/saml.ts

```diff
@@ -197,8 +197,12 @@ class SAML {
     }
   }
 
-  _generateUniqueID() {
-    return crypto.randomBytes(10).toString("hex");
+  private generateUniqueId() {
```

Comment:
I don't think that would be good idea, since if a specific wants that protection against accidentally using testing code in production, they could just add the env var check inside their custom generateUniqueId function?

We use the unique IDs as a channel to pass information through the sign-in flow, (so for example the corresponding 'inResponseTo' value from an ID provider can be used to determine which customer the request should be checked against)

Or, for example, it can include a shard key in the unique ID so that responses can be handled by separate servers without any other shared state.

Reply:
> We use the unique IDs as a channel to pass information through the sign-in flow

@autopulated your use case sounds like use case of SAML specification's`RelayState`.
See: https://github.com/node-saml/passport-saml/issues/541#issuecomment-788111948

Reply:
We also use RelayState for deep links, but it can be overridden by IDP configuration.

It's perhaps not an intended use of inResponseTo, but the way I see it, inResponseTo is intended to correlate responses to their original request for validation. And since our original requests might have come from different tenants, it made sense to me to use it to correlate the response with the correct tenant (at the same time as validating that it is a genuine response to an outbound request).